src/state/session/agent.ts

```diff
@@ -33,20 +33,30 @@ export async function createAgentAndResume(
     event: AtpSessionEvent,
   ) => void,
 ) {
-  const agent = new BskyAgent({service: storedAccount.service})
+  const agent = new BskyAgent({
+    service: storedAccount.service,
+    persistSession: event => {
+      const {session} = agent
+      if (!session?.did) return // @TODO SESSION - ?
+
+      onSessionChange(agent, session.did, event)
+      if (event !== 'create' && event !== 'update') {
+        addSessionErrorLog(session.did, event)
+      }
+    },
+  })
   if (storedAccount.pdsUrl) {
-    agent.pdsUrl = agent.api.xrpc.uri = new URL(storedAccount.pdsUrl)
+    agent.sessionManager.pdsUrl = new URL(storedAccount.pdsUrl)
   }
   const gates = tryFetchGates(storedAccount.did, 'prefer-low-latency')
   const moderation = configureModerationForAccount(agent, storedAccount)
   const prevSession: AtpSessionData = sessionAccountToSession(storedAccount)
   if (isSessionExpired(storedAccount)) {
     await networkRetry(1, () => agent.resumeSession(prevSession))
   } else {
-    agent.session = prevSession
     if (!storedAccount.signupQueued) {
-      // Intentionally not awaited to unblock the UI:
-      networkRetry(3, () => agent.resumeSession(prevSession)).catch(
+      // @TODO SESSION - We were not awaiting this before. Why? And why is it now broken if we do not await?
+      await networkRetry(3, () => agent.resumeSession(prevSession)).catch(
```

Comment:
What UI gets blocked here if we await? (Also, what changed that we need to await this now for it to work?)

Reply:
no longer necessary